code/web/src/modules/crate/Item.js

```diff
@@ -1,92 +1,113 @@
 // Imports
-import React, { PureComponent } from 'react'
-import PropTypes from 'prop-types'
-import { connect } from 'react-redux'
-import { Link, withRouter } from 'react-router-dom'
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Link, withRouter } from 'react-router-dom';
 
 // UI Imports
-import Card from '../../ui/card/Card'
-import Button from '../../ui/button/Button'
-import H4 from '../../ui/typography/H4'
-import Icon from '../../ui/icon'
-import { white, grey2, black } from '../../ui/common/colors'
+import Card from '../../ui/card/Card';
+import Button from '../../ui/button/Button';
+import H4 from '../../ui/typography/H4';
+import Icon from '../../ui/icon';
+import { white, grey2, black } from '../../ui/common/colors';
 
 // App Imports
-import { APP_URL } from '../../setup/config/env'
-import userRoutes from '../../setup/routes/user'
-import { messageShow, messageHide } from '../common/api/actions'
-import { create } from '../subscription/api/actions'
+import { APP_URL } from '../../setup/config/env';
+import userRoutes from '../../setup/routes/user';
+import { messageShow, messageHide } from '../common/api/actions';
+import { create } from '../subscription/api/actions';
 
 // Component
 class Item extends PureComponent {
-
   constructor(props) {
-    super(props)
+    super(props);
 
     this.state = {
-      isLoading: false
-    }
+      isLoading: false,
+    };
   }
 
   onClickSubscribe = (crateId) => {
     this.setState({
-      isLoading: true
-    })
+      isLoading: true,
+    });
 
-    this.props.messageShow('Subscribing, please wait...')
+    this.props.messageShow('Subscribing, please wait...');
 
-    this.props.create({ crateId })
-      .then(response => {
+    this.props
+      .create({ crateId })
+      .then((response) => {
         if (response.data.errors && response.data.errors.length > 0) {
-          this.props.messageShow(response.data.errors[0].message)
+          this.props.messageShow(response.data.errors[0].message);
         } else {
-          this.props.messageShow('Subscribed successfully.')
-
-          this.props.history.push(userRoutes.subscriptions.path)
+          this.props.messageShow('Subscribed successfully.');
+          if (this.props.user.details.style) {
+            this.props.history.push(userRoutes.subscriptions.path);
+          } else {
+            this.props.history.push(userRoutes.survey.path);
```

Comment:
I like the routing here, still getting the subscription for the user but then doing the routing logic